assets/src/edit-story/components/carousel/secondaryMenu.js

```diff
@@ -106,28 +110,42 @@ function SecondaryMenu() {
   // Only one popup is open at a time
   // we want to close an open popup if a new one is opened.
   useEffect(() => {
-    if (isChecklistOpen) {
+    if (isChecklistOpen && !isChecklistOpenRef.current) {
       closeHelpCenter();
       closeKeyboardShortcutsMenu();
+
+      isChecklistOpenRef.current = true;
+      isKeyboardShortcutsMenuOpenRef.current = false;
+      isHelpCenterOpenRef.current = false;
     }
   }, [closeHelpCenter, closeKeyboardShortcutsMenu, isChecklistOpen]);
 
   useEffect(() => {
-    if (isHelpCenterOpen) {
+    if (isHelpCenterOpen && !isHelpCenterOpenRef.current) {
       closeChecklist();
       closeKeyboardShortcutsMenu();
+      isHelpCenterOpenRef.current = true;
+      isChecklistOpenRef.current = false;
+      isKeyboardShortcutsMenuOpenRef.current = false;
     }
   }, [closeChecklist, closeKeyboardShortcutsMenu, isHelpCenterOpen]);
 
   useEffect(() => {
-    if (isKeyboardShortcutsMenuOpen) {
+    if (
+      isKeyboardShortcutsMenuOpen &&
+      !isKeyboardShortcutsMenuOpenRef.current
```

Comment:
These feel like redundant state 🤔 If we are adding a ref here I think we should remove the other state.

I think I'm still confused about why we would get stuck before 🤔

Reply:
It's because it's all async, so there's cases where one menu would still be open when another is asked for and it results in the asked for menu not actually expanding. The easiest way to see it is either in the before video or toggle between checklist and keyboard shortcuts on any other branch.

Reply:
Ah gotcha. So we need another piece of state just to tell us what the plan is with opening/closing panels